common/src/main/java/net/adoptopenjdk/icedteaweb/logging/LoggerFactory.java

```diff
@@ -6,14 +6,28 @@
 public class LoggerFactory {
 
     private static final String FACTORY_CLASS = "net.sourceforge.jnlp.util.logging.OutputControllerLoggerFactory";
-    private static final LoggerFactoryImpl factory = initFactory();
+    private static final LoggerFactoryImpl factory;
 
-    private static LoggerFactoryImpl initFactory() {
+    static {
+        Exception ex = null;
+        LoggerFactoryImpl loggerFactory;
         try {
             final Class<?> factoryClass = LoggerFactory.class.getClassLoader().loadClass(FACTORY_CLASS);
-            return (LoggerFactoryImpl) factoryClass.newInstance();
+            loggerFactory = (LoggerFactoryImpl) factoryClass.newInstance();
         } catch (Exception e) {
-            return new SystemOutLoggerFactory();
+            ex = e;
+            loggerFactory = new SystemOutLoggerFactory();
+        }
+        factory = loggerFactory;
+
+        // one can only get a logger after the factory has been set.
+        // therefore we postpone all the logging to after this comment
+
+        final Logger LOG = factory.getLogger(LoggerFactory.class);
+        if (ex != null) {
```

Comment:
if / else can be inlined in the try-catch

Reply:
tried this and failed. Reason is we cannot log before the static field `factory` is assigned. And assigning it in the try catch block does not compile.